// @azure/cognitiveservices-translatortext/esm/models/mappers.js
/*
 * Copyright (c) Microsoft Corporation. All rights reserved.
 * Licensed under the MIT License. See License.txt in the project root for license information.
 *
 * Code generated by Microsoft (R) AutoRest Code Generator.
 * Changes may cause incorrect behavior and will be lost if the code is regenerated.
 */
export var LanguagesResultTranslationLanguageCode = {
    serializedName: "LanguagesResult_translation_languageCode",
    type: {
        name: "Composite",
        className: "LanguagesResultTranslationLanguageCode",
        modelProperties: {
            name: {
                serializedName: "name",
                type: {
                    name: "String"
                }
            },
            nativeName: {
                serializedName: "nativeName",
                type: {
                    name: "String"
                }
            },
            dir: {
                serializedName: "dir",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export var LanguagesResultTranslation = {
    serializedName: "LanguagesResult_translation",
    type: {
        name: "Composite",
        className: "LanguagesResultTranslation",
        modelProperties: {
            languageCode: {
                serializedName: "languageCode",
                type: {
                    name: "Composite",
                    className: "LanguagesResultTranslationLanguageCode"
                }
            }
        }
    }
};
export var LanguagesResultTransliterationLanguageCodeScriptsItemToScriptsItem = {
    serializedName: "LanguagesResult_transliteration_languageCode_scriptsItem_toScriptsItem",
    type: {
        name: "Composite",
        className: "LanguagesResultTransliterationLanguageCodeScriptsItemToScriptsItem",
        modelProperties: {
            code: {
                serializedName: "code",
                type: {
                    name: "String"
                }
            },
            name: {
                serializedName: "name",
                type: {
                    name: "String"
                }
            },
            nativeName: {
                serializedName: "nativeName",
                type: {
                    name: "String"
                }
            },
            dir: {
                serializedName: "dir",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export var LanguagesResultTransliterationLanguageCodeScriptsItem = {
    serializedName: "LanguagesResult_transliteration_languageCode_scriptsItem",
    type: {
        name: "Composite",
        className: "LanguagesResultTransliterationLanguageCodeScriptsItem",
        modelProperties: {
            code: {
                serializedName: "code",
                type: {
                    name: "String"
                }
            },
            name: {
                serializedName: "name",
                type: {
                    name: "String"
                }
            },
            nativeName: {
                serializedName: "nativeName",
                type: {
                    name: "String"
                }
            },
            dir: {
                serializedName: "dir",
                type: {
                    name: "String"
                }
            },
            toScripts: {
                serializedName: "toScripts",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "LanguagesResultTransliterationLanguageCodeScriptsItemToScriptsItem"
                        }
                    }
                }
            }
        }
    }
};
export var LanguagesResultTransliterationLanguageCode = {
    serializedName: "LanguagesResult_transliteration_languageCode",
    type: {
        name: "Composite",
        className: "LanguagesResultTransliterationLanguageCode",
        modelProperties: {
            name: {
                serializedName: "name",
                type: {
                    name: "String"
                }
            },
            nativeName: {
                serializedName: "nativeName",
                type: {
                    name: "String"
                }
            },
            scripts: {
                serializedName: "scripts",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "LanguagesResultTransliterationLanguageCodeScriptsItem"
                        }
                    }
                }
            }
        }
    }
};
export var LanguagesResultTransliteration = {
    serializedName: "LanguagesResult_transliteration",
    type: {
        name: "Composite",
        className: "LanguagesResultTransliteration",
        modelProperties: {
            languageCode: {
                serializedName: "languageCode",
                type: {
                    name: "Composite",
                    className: "LanguagesResultTransliterationLanguageCode"
                }
            }
        }
    }
};
export var LanguagesResultDictionaryLanguageCodeTranslationsItem = {
    serializedName: "LanguagesResult_dictionary_languageCode_translationsItem",
    type: {
        name: "Composite",
        className: "LanguagesResultDictionaryLanguageCodeTranslationsItem",
        modelProperties: {
            name: {
                serializedName: "name",
                type: {
                    name: "String"
                }
            },
            nativeName: {
                serializedName: "nativeName",
                type: {
                    name: "String"
                }
            },
            dir: {
                serializedName: "dir",
                type: {
                    name: "String"
                }
            },
            code: {
                serializedName: "code",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export var LanguagesResultDictionaryLanguageCode = {
    serializedName: "LanguagesResult_dictionary_languageCode",
    type: {
        name: "Composite",
        className: "LanguagesResultDictionaryLanguageCode",
        modelProperties: {
            name: {
                serializedName: "name",
                type: {
                    name: "String"
                }
            },
            nativeName: {
                serializedName: "nativeName",
                type: {
                    name: "String"
                }
            },
            dir: {
                serializedName: "dir",
                type: {
                    name: "String"
                }
            },
            translations: {
                serializedName: "translations",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "LanguagesResultDictionaryLanguageCodeTranslationsItem"
                        }
                    }
                }
            }
        }
    }
};
export var LanguagesResultDictionary = {
    serializedName: "LanguagesResult_dictionary",
    type: {
        name: "Composite",
        className: "LanguagesResultDictionary",
        modelProperties: {
            languageCode: {
                serializedName: "languageCode",
                type: {
                    name: "Composite",
                    className: "LanguagesResultDictionaryLanguageCode"
                }
            }
        }
    }
};
export var LanguagesResult = {
    serializedName: "LanguagesResult",
    type: {
        name: "Composite",
        className: "LanguagesResult",
        modelProperties: {
            translation: {
                serializedName: "translation",
                type: {
                    name: "Composite",
                    className: "LanguagesResultTranslation"
                }
            },
            transliteration: {
                serializedName: "transliteration",
                type: {
                    name: "Composite",
                    className: "LanguagesResultTransliteration"
                }
            },
            dictionary: {
                serializedName: "dictionary",
                type: {
                    name: "Composite",
                    className: "LanguagesResultDictionary"
                }
            }
        }
    }
};
export var DictionaryExampleResultItemExamplesItem = {
    serializedName: "DictionaryExampleResultItem_examplesItem",
    type: {
        name: "Composite",
        className: "DictionaryExampleResultItemExamplesItem",
        modelProperties: {
            sourcePrefix: {
                serializedName: "sourcePrefix",
                type: {
                    name: "String"
                }
            },
            sourceTerm: {
                serializedName: "sourceTerm",
                type: {
                    name: "String"
                }
            },
            sourceSuffix: {
                serializedName: "sourceSuffix",
                type: {
                    name: "String"
                }
            },
            targetPrefix: {
                serializedName: "targetPrefix",
                type: {
                    name: "String"
                }
            },
            targetTerm: {
                serializedName: "targetTerm",
                type: {
                    name: "String"
                }
            },
            targetSuffix: {
                serializedName: "targetSuffix",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export var DictionaryExampleResultItem = {
    serializedName: "DictionaryExampleResultItem",
    type: {
        name: "Composite",
        className: "DictionaryExampleResultItem",
        modelProperties: {
            normalizedSource: {
                serializedName: "normalizedSource",
                type: {
                    name: "String"
                }
            },
            normalizedTarget: {
                serializedName: "normalizedTarget",
                type: {
                    name: "String"
                }
            },
            examples: {
                serializedName: "examples",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "DictionaryExampleResultItemExamplesItem"
                        }
                    }
                }
            }
        }
    }
};
export var DictionaryLookupResultItemTranslationsItemBackTranslationsItem = {
    serializedName: "DictionaryLookupResultItem_translationsItem_backTranslationsItem",
    type: {
        name: "Composite",
        className: "DictionaryLookupResultItemTranslationsItemBackTranslationsItem",
        modelProperties: {
            normalizedText: {
                serializedName: "normalizedText",
                type: {
                    name: "String"
                }
            },
            displayText: {
                serializedName: "displayText",
                type: {
                    name: "String"
                }
            },
            numExamples: {
                serializedName: "numExamples",
                type: {
                    name: "Number"
                }
            },
            frequencyCount: {
                serializedName: "frequencyCount",
                type: {
                    name: "Number"
                }
            }
        }
    }
};
export var DictionaryLookupResultItemTranslationsItem = {
    serializedName: "DictionaryLookupResultItem_translationsItem",
    type: {
        name: "Composite",
        className: "DictionaryLookupResultItemTranslationsItem",
        modelProperties: {
            normalizedTarget: {
                serializedName: "normalizedTarget",
                type: {
                    name: "String"
                }
            },
            displayTarget: {
                serializedName: "displayTarget",
                type: {
                    name: "String"
                }
            },
            posTag: {
                serializedName: "posTag",
                type: {
                    name: "String"
                }
            },
            confidence: {
                serializedName: "confidence",
                type: {
                    name: "Number"
                }
            },
            prefixWord: {
                serializedName: "prefixWord",
                type: {
                    name: "String"
                }
            },
            backTranslations: {
                serializedName: "backTranslations",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "DictionaryLookupResultItemTranslationsItemBackTranslationsItem"
                        }
                    }
                }
            }
        }
    }
};
export var DictionaryLookupResultItem = {
    serializedName: "DictionaryLookupResultItem",
    type: {
        name: "Composite",
        className: "DictionaryLookupResultItem",
        modelProperties: {
            normalizedSource: {
                serializedName: "normalizedSource",
                type: {
                    name: "String"
                }
            },
            displaySource: {
                serializedName: "displaySource",
                type: {
                    name: "String"
                }
            },
            translations: {
                serializedName: "translations",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "DictionaryLookupResultItemTranslationsItem"
                        }
                    }
                }
            }
        }
    }
};
export var TranslateResultItemTranslationItem = {
    serializedName: "TranslateResultItem_translationItem",
    type: {
        name: "Composite",
        className: "TranslateResultItemTranslationItem",
        modelProperties: {
            text: {
                serializedName: "text",
                type: {
                    name: "String"
                }
            },
            to: {
                serializedName: "to",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export var TranslateResultItem = {
    serializedName: "TranslateResultItem",
    type: {
        name: "Composite",
        className: "TranslateResultItem",
        modelProperties: {
            translation: {
                serializedName: "translation",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "TranslateResultItemTranslationItem"
                        }
                    }
                }
            }
        }
    }
};
export var TranslateResultAllItemDetectedLanguage = {
    serializedName: "TranslateResultAllItem_detectedLanguage",
    type: {
        name: "Composite",
        className: "TranslateResultAllItemDetectedLanguage",
        modelProperties: {
            language: {
                serializedName: "language",
                type: {
                    name: "String"
                }
            },
            score: {
                serializedName: "score",
                type: {
                    name: "Number"
                }
            }
        }
    }
};
export var TranslateResultAllItemTranslationsItemTransliteration = {
    serializedName: "TranslateResultAllItem_translationsItem_transliteration",
    type: {
        name: "Composite",
        className: "TranslateResultAllItemTranslationsItemTransliteration",
        modelProperties: {
            text: {
                serializedName: "text",
                type: {
                    name: "String"
                }
            },
            script: {
                serializedName: "script",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export var TranslateResultAllItemTranslationsItemAlignment = {
    serializedName: "TranslateResultAllItem_translationsItem_alignment",
    type: {
        name: "Composite",
        className: "TranslateResultAllItemTranslationsItemAlignment",
        modelProperties: {
            proj: {
                serializedName: "proj",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export var TranslateResultAllItemTranslationsItemSentLenSrcSentLenItem = {
    serializedName: "TranslateResultAllItem_translationsItem_sentLen_srcSentLenItem",
    type: {
        name: "Composite",
        className: "TranslateResultAllItemTranslationsItemSentLenSrcSentLenItem",
        modelProperties: {
            integer: {
                serializedName: "integer",
                type: {
                    name: "Number"
                }
            }
        }
    }
};
export var TranslateResultAllItemTranslationsItemSentLenTransSentLenItem = {
    serializedName: "TranslateResultAllItem_translationsItem_sentLen_transSentLenItem",
    type: {
        name: "Composite",
        className: "TranslateResultAllItemTranslationsItemSentLenTransSentLenItem",
        modelProperties: {
            integer: {
                serializedName: "integer",
                type: {
                    name: "Number"
                }
            }
        }
    }
};
export var TranslateResultAllItemTranslationsItemSentLen = {
    serializedName: "TranslateResultAllItem_translationsItem_sentLen",
    type: {
        name: "Composite",
        className: "TranslateResultAllItemTranslationsItemSentLen",
        modelProperties: {
            srcSentLen: {
                serializedName: "srcSentLen",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "TranslateResultAllItemTranslationsItemSentLenSrcSentLenItem"
                        }
                    }
                }
            },
            transSentLen: {
                serializedName: "transSentLen",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "TranslateResultAllItemTranslationsItemSentLenTransSentLenItem"
                        }
                    }
                }
            }
        }
    }
};
export var TranslateResultAllItemTranslationsItem = {
    serializedName: "TranslateResultAllItem_translationsItem",
    type: {
        name: "Composite",
        className: "TranslateResultAllItemTranslationsItem",
        modelProperties: {
            text: {
                serializedName: "text",
                type: {
                    name: "String"
                }
            },
            transliteration: {
                serializedName: "transliteration",
                type: {
                    name: "Composite",
                    className: "TranslateResultAllItemTranslationsItemTransliteration"
                }
            },
            to: {
                serializedName: "to",
                type: {
                    name: "String"
                }
            },
            alignment: {
                serializedName: "alignment",
                type: {
                    name: "Composite",
                    className: "TranslateResultAllItemTranslationsItemAlignment"
                }
            },
            sentLen: {
                serializedName: "sentLen",
                type: {
                    name: "Composite",
                    className: "TranslateResultAllItemTranslationsItemSentLen"
                }
            }
        }
    }
};
export var TranslateResultAllItem = {
    serializedName: "TranslateResultAllItem",
    type: {
        name: "Composite",
        className: "TranslateResultAllItem",
        modelProperties: {
            detectedLanguage: {
                serializedName: "detectedLanguage",
                type: {
                    name: "Composite",
                    className: "TranslateResultAllItemDetectedLanguage"
                }
            },
            translations: {
                serializedName: "translations",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Composite",
                            className: "TranslateResultAllItemTranslationsItem"
                        }
                    }
                }
            }
        }
    }
};
export var BreakSentenceResultItem = {
    serializedName: "BreakSentenceResultItem",
    type: {
        name: "Composite",
        className: "BreakSentenceResultItem",
        modelProperties: {
            sentLen: {
                serializedName: "sentLen",
                type: {
                    name: "Sequence",
                    element: {
                        type: {
                            name: "Number"
                        }
                    }
                }
            }
        }
    }
};
export var TransliterateResultItem = {
    serializedName: "TransliterateResultItem",
    type: {
        name: "Composite",
        className: "TransliterateResultItem",
        modelProperties: {
            text: {
                serializedName: "text",
                type: {
                    name: "String"
                }
            },
            script: {
                serializedName: "script",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export var DetectResultItem = {
    serializedName: "DetectResultItem",
    type: {
        name: "Composite",
        className: "DetectResultItem",
        modelProperties: {
            text: {
                serializedName: "text",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export var BreakSentenceTextInput = {
    serializedName: "BreakSentenceTextInput",
    type: {
        name: "Composite",
        className: "BreakSentenceTextInput",
        modelProperties: {
            text: {
                serializedName: "text",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export var ErrorMessageError = {
    serializedName: "ErrorMessage_error",
    type: {
        name: "Composite",
        className: "ErrorMessageError",
        modelProperties: {
            code: {
                serializedName: "code",
                type: {
                    name: "String"
                }
            },
            message: {
                serializedName: "message",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export var ErrorMessage = {
    serializedName: "ErrorMessage",
    type: {
        name: "Composite",
        className: "ErrorMessage",
        modelProperties: {
            error: {
                serializedName: "error",
                type: {
                    name: "Composite",
                    className: "ErrorMessageError"
                }
            }
        }
    }
};
export var DetectTextInput = {
    serializedName: "DetectTextInput",
    type: {
        name: "Composite",
        className: "DetectTextInput",
        modelProperties: {
            text: {
                serializedName: "text",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export var DictionaryLookupTextInput = {
    serializedName: "DictionaryLookupTextInput",
    type: {
        name: "Composite",
        className: "DictionaryLookupTextInput",
        modelProperties: {
            text: {
                serializedName: "text",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export var DictionaryExampleTextInput = {
    serializedName: "DictionaryExampleTextInput",
    type: {
        name: "Composite",
        className: "DictionaryExampleTextInput",
        modelProperties: {
            text: {
                serializedName: "text",
                type: {
                    name: "String"
                }
            },
            translation: {
                serializedName: "translation",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export var TranslateTextInput = {
    serializedName: "TranslateTextInput",
    type: {
        name: "Composite",
        className: "TranslateTextInput",
        modelProperties: {
            text: {
                serializedName: "text",
                type: {
                    name: "String"
                }
            }
        }
    }
};
export var TransliterateTextInput = {
    serializedName: "TransliterateTextInput",
    type: {
        name: "Composite",
        className: "TransliterateTextInput",
        modelProperties: {
            text: {
                serializedName: "text",
                type: {
                    name: "String"
                }
            }
        }
    }
};
